pkg/gadgets/ebpftop/piditer/iter.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package piditer
+
+// #include <linux/types.h>
+// #include "./bpf/pid_iter.h"
+import "C"
```

Comment:
You can ask bpf2go to generate a Go struct equivalent to `C.struct_pid_iter_entry` so you don't need to have `import "C"` here:

```diff
--- a/pkg/gadgets/ebpftop/piditer/bpf/pid_iter.bpf.c
+++ b/pkg/gadgets/ebpftop/piditer/bpf/pid_iter.bpf.c
@@ -8,8 +8,14 @@
 
 const volatile __u64 bpf_prog_fops_addr = 0;
 
+/**
+ * Emit struct type info into the ELF's BTF so bpf2go 
+ * can generate a Go type from it. 
+ */ 
+struct pid_iter_entry *unused __attribute__((unused));
+
 SEC("iter/task_file")
--- a/pkg/gadgets/ebpftop/piditer/iter.go
+++ b/pkg/gadgets/ebpftop/piditer/iter.go
@@ -34,7 +34,7 @@ import (
        "golang.org/x/sys/unix"
 )
 
-//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target $TARGET -cc clang piditer ./bpf/pid_iter.bpf.c -- -I./bpf/ -I../../../${TARGET}
+//go:generate go run -mod vendor github.com/cilium/ebpf/cmd/bpf2go -target $TARGET -cc clang -type pid_iter_entry piditer ./bpf/pid_iter.bpf.c -- -I./bpf/ -I../../../${TARGET}
 
 type pidIter struct {
        objs piditerObjects
```

Then piditer_bpfel_x86.go will contain the following:
```
type piditerPidIterEntry struct {
       Id   uint32
       Pid  uint32
       Comm [16]int8
}
```

Reply:
Changed, thanks!